os/fs/smartfs/smartfs_smart.c

```diff
@@ -783,7 +783,11 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 				fdbg("Error %d writing sector %d data\n", ret, sf->currsector);
 				goto errout_with_semaphore;
 			}
-
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
```

Comment:
Adding this code after BIOC_WRITESECT has been called means
bytes are copied from buffer[byteswritten] to sf->buffer for synchronizing later.

But, with WRITESECT, bytes from position buffer[byteswritten] have already been written to MTD.

So either they should be written or else copied for writing later if CRC is enabled, not both.

a. > #ifndef CONFIG_SMARTFS_USE_SECTOR_BUFFER
    > BIOC_WRITESECT (Line 781 - 785)
    > #else 
    > memcpy to sf->buffer

OR

b. > BIOC_WRITESECT
    > byteswritten += readwrite.count
    > memcpy to sf->buffer (copy next set of bytes)

Should be done in one of the above ways.

Reply:
You had pass my quiz, I wanted answer 'a' =D
Yes. current code will bring 2times of write.

Reply:
@amogh-samsung  I thought same as you but I faced another situation, will be modifed more.

Reply:
updated.

Reply:
Ohkkay now both copy to sf buffer and write to MTD will not happen as I had mentioned.

But why for the last sector, smartfs_sync_internal is chosen for writing?

Reply:
yes because it has nextsector, working through the loop one more time.